src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,43 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
+import { Status as StatusType } from '../../types/StatusEnum';
 
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const setQuery = (query: string) => dispatch(filterActions.setQuery(query));
+  const setStatus = (status: Status) =>
+    dispatch(filterActions.setStatus(status));
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setStatus(event.target.value as Status);
+  };
+
+  const handleCLear = () => {
+    setQuery('');
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            onChange={handleSelect}
+            value={filter.status}
+          >
+            <option value={StatusType.All}>{StatusType.All}</option>
+            <option value={StatusType.Active}>{StatusType.Active}</option>
+            <option value={StatusType.Completed}>{StatusType.Completed}</option>
```

Comment:
Consider will be better if you create array with these values and mapping it here